app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/OwnedListFragment.java

```diff
@@ -196,53 +212,17 @@ private void onViewRequests(String bookId) {
         startActivity(intent);
     }
 
-    /**
-     * when the user click on the filter button
-     * @param view the filter button view
-     */
-    private void showFilterPopup(View view) {
-        LayoutInflater inflater = (LayoutInflater) getActivity().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
-        View layout = inflater.inflate(R.layout.owned_filter_menu, getActivity().findViewById(R.id.popup_element));
-        layout.measure(View.MeasureSpec.UNSPECIFIED, View.MeasureSpec.UNSPECIFIED);
-        int width = layout.getMeasuredWidth();
-        int height = layout.getMeasuredHeight();
-        PopupWindow popup = new PopupWindow(layout, width, height,true);
-
-        SwitchMaterial availableStatusSwitch = layout.findViewById(R.id.filter_menu_available);
-        SwitchMaterial requestedStatusSwitch = layout.findViewById(R.id.filter_menu_requested);
-        SwitchMaterial acceptedStatusSwitch = layout.findViewById(R.id.filter_menu_accepted);
-        SwitchMaterial borrowedStatusSwitch = layout.findViewById(R.id.filter_menu_borrowed);
-        availableStatusSwitch.setOnClickListener(this::onFilter);
-        requestedStatusSwitch.setOnClickListener(this::onFilter);
-        acceptedStatusSwitch.setOnClickListener(this::onFilter);
-        borrowedStatusSwitch.setOnClickListener(this::onFilter);
-        availableStatusSwitch.setChecked(includeAvailable);
-        requestedStatusSwitch.setChecked(includeRequested);
-        acceptedStatusSwitch.setChecked(includeAccepted);
-        borrowedStatusSwitch.setChecked(includeBorrowed);
-
-        // elevation does not work in xml file so have to set it here
-        popup.setElevation(12);
-        // want to show the window above the view instead of below
-        // so offset the window by its width and height
-        popup.showAsDropDown(view, -width, -height);
-    }
-
     private void onFilter(View view) {
         View parentView = (View) view.getParent();
         SwitchMaterial availableStatusSwitch = parentView.findViewById(R.id.filter_menu_available);
         SwitchMaterial requestedStatusSwitch = parentView.findViewById(R.id.filter_menu_requested);
         SwitchMaterial acceptedStatusSwitch = parentView.findViewById(R.id.filter_menu_accepted);
         SwitchMaterial borrowedStatusSwitch = parentView.findViewById(R.id.filter_menu_borrowed);
```

Comment:
these views are defined in onCreateView, too. Are we able to make them class attributes and make sure they only initialize once instead of every listener call?